static/js/message_edit.js

```diff
@@ -67,6 +67,11 @@ function get_editability(message, edit_limit_seconds_buffer = 0) {
     if (!is_topic_editable(message, edit_limit_seconds_buffer)) {
         return editability_types.NO;
     }
+
+    // Right now, we prevent users from editing widgets.
+    if (message.submessages && message.submessages.length !== 0) {
+        return editability_types.TOPIC_ONLY;
```

Comment:
This skips over all the below reasons that the message might not be editable at all: `locally_echoed`, `!realm_allow_message_editing`, `currently_echoing_messages`, `type !== "stream"`.

Reply:
How should I go about this?
If I put the submessage check below them and the organization has set `realm_message_content_edit_limit_seconds` to 0, the message will become editable.
(Although, it doesn't get updated at the backend.)

Also, I didn't quite understand why we allow editing messages if `realm_message_content_edit_limit_seconds` is set to 0. Is it for network issues?

And for `type !== "stream"`, this seems to run correctly.

Reply:
There’s no rule that says this function must be written as an ordered linear sequence of `if (condition) { return value; }` statements. The desired behavior can’t be expressed that way.

As you’ve written it now, an earlier `FULL` can incorrectly override your `TOPIC_ONLY`, and your `TOPIC_ONLY` can incorrectly override a later `NO_LONGER`.

Reply:
@andersk Made the changes. Would you please review this?